sdk/core/azure-core-test/src/main/java/com/azure/core/test/policy/RecordNetworkCallPolicy.java

```diff
@@ -102,40 +123,77 @@ public RecordNetworkCallPolicy(RecordedData recordedData) {
             responseData.put("retry-after", "0");
         }
 
-        String contentType = response.headerValue("content-type");
+        String contentType = response.headerValue("Content-Type");
         if (contentType == null) {
             return Mono.just(responseData);
-        } else if (contentType.contains("json") || response.headerValue("content-encoding") == null) {
-            return response.bodyAsString().switchIfEmpty(Mono.just("")).map(content -> {
-                responseData.put("Body", content);
+        } else if (contentType.contains("octet-stream")) {
```

Comment:
There are a lot of special string literals "octet-stream", "Body", "Content-Encoding". It would be nice to consolidate these in either a constants file or at the top of the class so we don't miss updating one.

Reply:
Same comment about application/octet-stream

Reply:
Updated